app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt

```diff
@@ -329,6 +341,25 @@ class DeadlineDetailsActivity : AppCompatActivity() {
         }
     }
 
+    // Set the information of the notifications given by the deadline.
+    // If there are more notifications than the default ones, the function
+    // add them on the arrays to be able to show them in the Dialog.
+    private fun setNotifications(notifications: List<Long>) {
+        for (time in notifications) {
+            if (timeNotifications.contains(time)) {
+                notificationSelected[timeNotifications.indexOf(time)] = true
```

Comment:
It is very probably not a problem here as no two elements in `notifications` have the same value, but be careful with the use of `indexOf`. Prefer a pattern like:
```Kotlin
for ((idx, time) in notifications.withIndex()){
     ...
             notificationsSelected[idx] = true
     ...
}
```